src/client/src/producer.rs

```diff
@@ -56,6 +56,12 @@ impl TopicProducer {
 
         send_record_raw(spu_client, &replica, record).await
     }
+
+    /// Returns true if a simple connection to `self.topic/0` is successful
+    pub async fn topic_exists(&self) -> bool {
+        let replica = ReplicaKey::new(&self.topic, 0);
+        self.pool.create_serial_socket(&replica).await.is_ok()
```

Comment:
can you move this to `SpuPool` ?  In that case, can you use `lookup_by_key` and not have to create socket